src/main/java/uk/gov/hmcts/reform/unspec/service/flowstate/StateFlowEngine.java

```diff
@@ -70,81 +85,82 @@
-                .transitionTo(CLAIM_WITHDRAWN).onlyIf(claimWithdrawn)
-                .transitionTo(CASE_PROCEEDS_IN_CASEMAN).onlyIf(caseProceedsInCaseman)
-            .state(RESPONDENT_FULL_DEFENCE)
+            .state(CLAIM_DETAILS_NOTIFIED)
+                .transitionTo(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION).onlyIf(claimDetailsNotifiedTimeExtension)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged)
+                .transitionTo(FULL_DEFENCE).onlyIf(fullDefenceAfterNotifyDetails)
+                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmissionAfterNotifyDetails)
+                .transitionTo(PART_ADMISSION).onlyIf(partAdmissionAfterNotifyDetails)
+                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaimAfterNotifyDetails)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF).onlyIf(takenOfflineByStaffAfterClaimDetailsNotified)
+                .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE).onlyIf(caseDismissed)
+            .state(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledgedTimeExtension)
+                .transitionTo(FULL_DEFENCE).onlyIf(fullDefence)
+                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmission)
+                .transitionTo(PART_ADMISSION).onlyIf(partAdmission)
+                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaim)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF).onlyIf(takenOfflineByStaffAfterClaimDetailsNotifiedExtension)
+            .state(NOTIFICATION_ACKNOWLEDGED)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED_TIME_EXTENSION)
+                    .onlyIf(notificationAcknowledgedTimeExtension)
+                .transitionTo(FULL_DEFENCE).onlyIf(fullDefenceAfterAcknowledge)
+                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmissionAfterAcknowledge)
+                .transitionTo(PART_ADMISSION).onlyIf(partAdmissionAfterAcknowledge)
+                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaimAfterAcknowledge)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF)
+                    .onlyIf(takenOfflineByStaffAfterNotificationAcknowledged)
+                .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE).onlyIf(caseDismissedAfterClaimAcknowledged)
+            .state(NOTIFICATION_ACKNOWLEDGED_TIME_EXTENSION)
+                .transitionTo(FULL_DEFENCE).onlyIf(fullDefence)
+                .transitionTo(FULL_ADMISSION).onlyIf(fullAdmission)
+                .transitionTo(PART_ADMISSION).onlyIf(partAdmission)
+                .transitionTo(COUNTER_CLAIM).onlyIf(counterClaim)
+                .transitionTo(TAKEN_OFFLINE_BY_STAFF)
+                    .onlyIf(takenOfflineByStaffAfterNotificationAcknowledgedTimeExtension)
+                .transitionTo(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE).onlyIf(caseDismissedAfterClaimAcknowledged)
```

Comment:
This transition will be at error state. as using the same predicate `caseDismissedAfterClaimAcknowledged` used for previous state. Below is the test case to verify that.

```@Test
        void shouldReturnClaimDismissedPastDeadline_whenCaseDataAtStateNotificationAcknowledgedTimeExtension() {
            CaseData caseData = CaseDataBuilder.builder().atStateNotificationAcknowledgedTimeExtension()
                .claimDismissedDate(LocalDateTime.now())
                .build();

            StateFlow stateFlow = stateFlowEngine.evaluate(caseData);

            assertThat(stateFlow.getState())
                .extracting(State::getName)
                .isNotNull()
                .isEqualTo(CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE.fullName());
            assertThat(stateFlow.getStateHistory())
                .hasSize(9)
                .extracting(State::getName)
                .containsExactly(
                    DRAFT.fullName(),
                    CLAIM_SUBMITTED.fullName(), CLAIM_ISSUED_PAYMENT_SUCCESSFUL.fullName(),
                    PENDING_CLAIM_ISSUED.fullName(), CLAIM_ISSUED.fullName(), CLAIM_NOTIFIED.fullName(),
                    CLAIM_DETAILS_NOTIFIED.fullName(), NOTIFICATION_ACKNOWLEDGED.fullName(),
                    NOTIFICATION_ACKNOWLEDGED_TIME_EXTENSION.fullName(),
                    CLAIM_DISMISSED_PAST_CLAIM_DISMISSED_DEADLINE.fullName()
                );
        }
        ````